fedbiomed/common/config.py

```diff
@@ -297,8 +235,78 @@ def refresh(self):
             )
 
         # Read the config
-        self._cfg.read(self.path)
+        self._cfg.read(self.config_path)
         id = self._cfg["default"]["id"]
 
         # Generate by keeping the component ID
         self.generate(force=True, id=id)
+
+
+class Component:
+
+    config_cls: type
+    _config: Config
+    _default_component_name: str
+
+    def __init__(self):
+        """Test"""
+        self._reference = '.fedbiomed'
+
+    def create(self, root: str | None = None) -> Config:
+        """Create component"""
+
+        if not root:
+            root = os.path.join(os.getcwd(), self._default_component_name)
+
+        reference = self.validate(root)
+        config = self.config_cls(root, auto_generate=False)
+
+        if not os.path.isfile(reference):
+            create_fedbiomed_setup_folders(root)
+            with open(os.path.join(root, '.fedbiomed'), 'w', encoding='UTF-8') as file_:
+                file_.write(self.config_cls.COMPONENT_TYPE)
+            config.generate()
+            config.write()
+        else:
+            config.read()
+
+        return config
+
+
+    def is_component_existing(self, component_dir: str) -> bool:
+        """Checks if component existing in the given root directory
+
+        Returns:
+            True if any component is instantiated in the given directory
+        """
+        ref = os.path.join(component_dir, self._reference)
+        if os.path.isdir(component_dir):
+            if os.listdir(component_dir) and not os.path.isfile(ref):
+                raise ValueError(
+                    f'Path {component_dir} is not empty for Fed-BioMed component initialization.'
+                )
```

Comment:
In the error message, I would have been more specific, by displaying:` f'cannot create component: {component}. Path {component_dir} is not empty for Fed-BioMed component initialization. Please remove folder {component_dir} or specify another path'`

Reply:
Thanks yannick.  I updated the message.